lib/sycamore/sycamore/transforms/merge_elements.py

```diff
@@ -135,7 +135,151 @@ def merge(self, elt1: Element, elt2: Element) -> Element:
 
         return new_elt
 
+class GreedySectionMerger(ElementMerger):
+    def __init__(self, tokenizer: Tokenizer, max_tokens: int, merge_across_pages: bool = True):
+        self.tokenizer = tokenizer
+        self.max_tokens = max_tokens
+        self.merge_across_pages = merge_across_pages
+
+    def preprocess_element(self, element: Element) -> Element:
+        if element.type == 'Image' and "summary" in element.properties:
+            element.data["token_count"] = len(self.tokenizer.tokenize(element.properties['summary'] or ""))
+        else:
+            element.data["token_count"] = len(self.tokenizer.tokenize(element.text_representation or ""))
+        return element
+
+    def postprocess_element(self, element: Element) -> Element:
+        del element.data["token_count"]
+        return element
+
+    def should_merge(self, element1: Element, element2: Element) -> bool:
+        # deal with empty elements
+        if "page_number" not in element1.properties or "page_number" not in element2.properties or element1.type == None or element2.type == None:
+            return False
+
+        # DO NOT MERGE across pages
+        if not self.merge_across_pages and element1.properties["page_number"] != element2.properties["page_number"]:
+            return False
+
+        if element1.data["token_count"] + 1 + element2.data["token_count"] > self.max_tokens:
+            return False
+        
+        # MERGE adjacent 'text' elements (but not across pages - see above)
+        if ((element1 != None) and (element1.type == 'Text') and 
```

Comment:
we should just fail/return early if the element is None right?

Reply:
Yes, this is made superfluous by the test in line 157. Will remove